src/logic/safe/transactions/gasNew.js

```diff
@@ -58,3 +82,33 @@ export const estimateTxGasCosts = async (
     return 10000
   }
 }
+
+export const estimateSafeTxGas = async (
+  safe: any,
+  safeAddress: string,
+  data: string,
+  to: string,
+  valueInWei: number | string,
+  operation: number,
+) => {
+  try {
+    let safeInstance = safe
+    if (!safeInstance) {
+      safeInstance = await getGnosisSafeInstanceAt(safeAddress)
+    }
+
+    const estimateData = safeInstance.contract.methods.requiredTxGas(to, valueInWei, data, operation).encodeABI()
+    const estimateResponse = await getWeb3().eth.call({
+      to: safeAddress,
+      from: safeAddress,
+      data: estimateData,
+    })
+    const txGasEstimate = new BigNumber(estimateResponse.substring(138), 16)
+
+    // Add 10k else we will fail in case of nested calls
+    return txGasEstimate.toNumber() + 10000
```

Comment:
So we encounter some issues with nested calls were the 10k where not enough. @germartinez and @Uxio0 implemented solutions for the CPK and relay service. There is also a PR on the contracts repo to improve tests related to that https://github.com/gnosis/safe-contracts/pull/188

Reply:
For reference https://github.com/gnosis/gnosis-py/commit/ecaad24f368c91ad743bb863b28888b336195c15